adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -213,6 +214,16 @@ private void performAuthorityValidation(final AuthenticationRequest authenticati
                 try {
                     mDiscovery.validateAuthority(authorityUrl);
                 } catch (final AuthenticationException authenticationException) {
+                    if (null != authenticationException.getCode()
+                            && (authenticationException.getCode().equals(ADALError.DEVICE_CONNECTION_IS_NOT_AVAILABLE)
+                            || authenticationException.getCode().equals(ADALError.NO_NETWORK_CONNECTION_POWER_OPTIMIZATION))) {
+                        Logger.w(TAG, "Connection is not available to validate the authority.", "", authenticationException.getCode());
+                        apiEvent.setValidationStatus(EventStrings.AUTHORITY_VALIDATION_FAILURE);
+                        Telemetry.getInstance().stopEvent(authenticationRequest.getTelemetryRequestId(), apiEvent,
+                                EventStrings.AUTHORITY_VALIDATION_EVENT);
+                        throw authenticationException;
```

Comment:
@iambmelt could you help me to check if the Telemetry event is processed correctly? Thanks.

Reply:
@addev-ashish Previously, all authentication exceptions are ignored here. I am not very clear the previous intention of the implementation here. However, I think the authentication exception should not be ignored if the authority validation failed with network connection error. Instead, the authentication exception with network error code should be thrown to the calling app.

Reply:
Do we have a SPEC here to explain why we ignore the authority validation failure here?

Reply:
I am going to remove this part, we should let the flow continue if the validateAuthority is turned off.